src/concrete/ml/sklearn/xgb.py

```diff
@@ -159,6 +179,15 @@ def dump_dict(self) -> Dict[str, Any]:
         metadata["use_label_encoder"] = self.use_label_encoder
         metadata["random_state"] = self.random_state
         metadata["verbosity"] = self.verbosity
+        metadata["max_bin"] = self.max_bin
+        metadata["callbacks"] = self.callbacks
```

Comment:
great, but I'm not sure we support some of these parameters when it comes to serialization. for example for QNNs we have this :

```
        # skorch attributes that cannot be serialized
        # FIXME: https://github.com/zama-ai/concrete-ml-internal/issues/3550
        # Disable mypy as running isinstance with a Callable type unexpectedly raises an issue:
        # https://github.com/python/mypy/issues/3060
        if isinstance(self.train_split, Callable) and not isinstance(  # type: ignore[arg-type]
            self.train_split, ValidSplit
        ):
            raise NotImplementedError(
                "Serializing a custom Callable object is not secure and is therefore disabled. "
                "Please set `train_split` to either None or a ValidSplit instance."
            )

        if self.callbacks != "disable":
            raise NotImplementedError(
                "Serializing a custom Callable object is not secure and is therefore disabled. "
                "Additionally, the serialization of skorch's different callback classes is not "
                f"supported. Please set `callbacks` to 'disable'. Got {self.callbacks}."
            )
```

Reply:
Hmm don't we have a test that checks we can serailize and desarialize all parameters? Tests seem to pass.

Reply:
yes but it uses the default values, and we are able to serialize a None 🙂  here I'm talking about serializing exotic stuff like `TrainingCallback` (for `self.callbacks`) or callable (ex: `self. eval_metric`) objects that a user could put when initializing

Reply:
I see :/ I will assert potential callable parameter are None. eval_metric isn't callable it's a string though, right?

Reply:
better to have a value error raised + tested like we do for QNNs ! putting asserts for user inputs is not good a good practice (refs https://github.com/zama-ai/concrete-ml-internal/issues/4232) 

besides about `eval_metric`, I can see that you've put `eval_metric: Optional[Union[str, List[str], Callable]] = None,` , so I imagine it can also be a `callable`

Reply:
> besides about eval_metric, I can see that you've put eval_metric: Optional[Union[str, List[str], Callable]] = None, , so I imagine it can also be a callable

Hmm I see. I just copy pasted the one from XGBoostClassifier. Looks like the XGBoostRegressor type is wrong then.

Maybe we should also check the types of the params. It's starting to be quite heavy lol

Reply:
I added the raising of errors.